.github/workflows/multibuild.yaml

```diff
@@ -59,18 +64,70 @@ jobs:
         run: dart compile exe bin/srvd.dart -v -o sshnp/srvd${{ matrix.ext }}
       - if: ${{ matrix.os != 'windows-latest' }}
         run: dart compile exe bin/srvd.dart -D ENABLE_SNOOP=true -v -o sshnp/debug/srvd${{ matrix.ext }}
+      # copy additional bundle items to build
       - run: cp -r bundles/core/* sshnp/
       - run: cp -r bundles/${{ matrix.bundle }}/* sshnp/
       - run: cp LICENSE sshnp
-      - run: tar -cvzf tarball/${{ matrix.output-name }}.tgz sshnp
+      # codesign for apple
+      - if: ${{ matrix.os == 'macOS-latest' || matrix.os == 'macos-14'}}
+        name: Import certificates
+        env:
+          MACOS_CODESIGN_CERT: ${{ secrets.MACOS_CODESIGN_CERT }}
+          MACOS_CODESIGN_CERT_PASSWORD: ${{ secrets.MACOS_CODESIGN_CERT_PASSWORD }}
+          MACOS_SIGNING_IDENTITY: ${{ secrets.MACOS_SIGNING_IDENTITY }}
+          MACOS_KEYCHAIN_PASSWORD: ${{ secrets.MACOS_KEYCHAIN_PASSWORD }}
+        run: |
+          # Load certificate
+          CERT_PATH=$RUNNER_TEMP/noports-codesign.p12
+          echo -n "$MACOS_CODESIGN_CERT" | base64 --decode -o $CERT_PATH
+          # create temp keychain
+          KEYCHAIN_PATH=$RUNNER_TEMP/build.keychain
+          security create-keychain -p "$MACOS_KEYCHAIN_PASSWORD" $KEYCHAIN_PATH
+          security default-keychain -s $KEYCHAIN_PATH
+          security unlock-keychain -p "$MACOS_KEYCHAIN_PASSWORD" $KEYCHAIN_PATH
+          security import $CERT_PATH -k $KEYCHAIN_PATH -P "$MACOS_CODESIGN_CERT_PASSWORD" -T /usr/bin/codesign
+          security set-key-partition-list -S apple-tool:apple,:,codesign: -s -k "$MACOS_KEYCHAIN_PASSWORD" $KEYCHAIN_PATH
+ 
+          # codesign
+          /usr/bin/codesign \
+            --force \
+            -s "$MACOS_SIGNING_IDENTITY" \
+            --options=runtime \
+            --entitlements ./tools/templates/entitlements.plist \
+            --prefix "com.atsign." \
+            --timestamp \
+            -v \
+            sshnp/{ssh*,srv,srvd,at_activate,debug/srvd}
+
+      # zip the build
+      - if: ${{ matrix.os == 'macOS-latest' || matrix.os == 'macos-14'}}
+        run: ditto -c -k --keepParent sshnp tarball/${{ matrix.output-name }}.zip
+      - if: ${{ matrix.os == 'ubuntu-latest' }}
+        run: tar -cvzf tarball/${{ matrix.output-name }}.tgz sshnp
+      - if: ${{ matrix.os == 'windows-latest' }}
+        run: Compress-Archive -Path sshnp -Destination tarball/${{ matrix.output-name }}.zip
+      # Notarize the build
+      - if: ${{ matrix.os == 'macOS-latest' || matrix.os == 'macos-14'}}
+        env:
+          MACOS_APPLE_ID: ${{ secrets.MACOS_APPLE_ID }}
+          MACOS_TEAM_ID: ${{ secrets.MACOS_TEAM_ID }}
+          MACOS_APPLE_ID_PASSWORD: ${{ secrets.MACOS_APPLE_ID_PASSWORD }}
```

Comment:
Team id is the Atsign Company team's ID

Apple ID password is a special generated app password, not your user login